modules/apps/frontend-js/frontend-js-test/src/testFunctional/tests/SessionAutomationExtension.testcase

```diff
@@ -0,0 +1,36 @@
+@component-name = "portal-frontend-infrastructure"
+definition {
+
+	property custom.properties = "session.timeout=2${line.separator}session.timeout.auto.extend.offset=115";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.main.component.name = "Login";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			User.firstLoginPG();
+		}
+	}
+
+	@description = "LPS-151640. Validates periodic requests without logging in."
+	@priority = "3"
+	test CanViewPeriodicRequestsWithoutLoggingIn {
+		task ("Then not view expired session error message") {
+			User.viewSessionNotExpired(noSignin = "True");
+		}
+
+		task ("Then assert periodic extend session requests") {
+			CheckPeriodicExtendSessionRequests.getPeriodicExtendSessionRequests();
```

Comment:
From what I recall, session.timeout.auto.extend.offset=115 will auto extend the session every 115 seconds. So, my question is, how will this check that the session has been auto extended after 115 seconds?

Reply:
I think if not message after 115 seconds, it's right like this
`task ("Then not view expired session error message") {
			User.viewSessionNotExpired(noSignin = "True");
		}`
May be I need add a `pause(locator1 = “115000”);` before view message

Reply:
sorry, I recalled incorrectly. session.timeout=2 means the session will timeout after 2 minutes, session.timeout.auto.extend.offset=115 means it will auto extend 115 seconds before session timeout. So it will auto extend every 5 seconds. 2 minutes - 115 seconds. Perhaps pause every 5 seconds before view message.